dubbo-metadata/dubbo-metadata-report-redis/src/main/java/org/apache/dubbo/metadata/store/redis/RedisMetadataReport.java

```diff
@@ -61,12 +77,34 @@ public class RedisMetadataReport extends AbstractMetadataReport {
     private Set<HostAndPort> jedisClusterNodes;
     private int timeout;
     private String password;
+    private final String root;
+    private MD5Utils md5Utils = new MD5Utils();
+    protected ConcurrentMap<String, MappingDataListener> listenerMap = new ConcurrentHashMap<>();
+    private String luaScript="local key = KEYS[1];\n" +
+        "local field = ARGV[1];\n" +
+        "local oldValue = ARGV[2];\n" +
+        "local newValue = ARGV[3];\n" +
+        "local channel = ARGV[4];\n" +
+        "local valueExists = redis.call(\"HEXISTS\", key, field);\n" +
+        "if valueExists == 0 then\n" +
+        "    redis.call(\"HSET\", key, field, newValue);\n" +
```

Comment:
Starting from version 2.2, redis provides `WATCH` to help implement [cas behavior](https://redis.io/docs/interact/transactions/#cas) in case race conditions may occur.

Reply:
请问下JedisCluster中无法使用watch有什么好的建议呢？
如果不建议使用lua脚本的话，要不然使用setnx实现一个分布式锁？

Reply:
之前有人[讨论](https://groups.google.com/g/jedis_redis/c/Z4-0065UFkg)过在JedisCluster上如何做transaction，基本思路是从sluter里拿到对应key所在的Jedis，然后在上面开transaction，只不过这样做的话会稍微麻烦点。
我觉得用什么方式都无所谓，只要能实现目标就行。

Reply:
不错的建议，PTAL

Reply:
麻烦再改一下watch那里的逻辑，其他部分我觉得ok。

Reply:
已改，之前完全没有注意过UNWATCH，十分感谢